aptos-move/framework/src/natives/network_address.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use aptos_types::network_address::NetworkAddress;
+use move_deps::{
+    move_binary_format::errors::PartialVMResult,
+    move_core_types::gas_algebra::{InternalGas, InternalGasPerByte, NumBytes},
+    move_vm_runtime::native_functions::{NativeContext, NativeFunction},
+    move_vm_types::{
+        loaded_data::runtime_types::Type, natives::function::NativeResult, pop_arg, values::Value,
+    },
+};
+use smallvec::smallvec;
+use std::{collections::VecDeque, sync::Arc};
+
+#[derive(Debug, Clone)]
+pub struct NetworkAddressGasParameters {
+    pub base: InternalGas,
+    pub per_byte: InternalGasPerByte,
+}
+
+fn native_validate_network_addresses(
+    gas_params: &NetworkAddressGasParameters,
+    _context: &mut NativeContext,
+    mut _ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    debug_assert!(_ty_args.is_empty());
+    debug_assert!(args.len() == 1);
+
+    let bytes = pop_arg!(args, Vec<u8>);
+    let cost = gas_params.base + gas_params.per_byte * NumBytes::new(bytes.len() as u64);
+    let is_valid = bcs::from_bytes::<Vec<NetworkAddress>>(&bytes).is_ok();
```

Comment:
This is currently invalid, unless there's a deserialization I'm missing.

If you look at the code the following happens:
1. Create `Vec<NetworkAddress>`
2. `encoded = bcs::to_bytes(Vec<NetworkAddresses>)`
3. `encoded_twice = bcs::to_bytes(encoded)`

This lets us put the object into a `Vec<u8>`

Reply:
At least so far this has worked in unit tests with validator/network addresses pulled from the current AIT3. For example 012d04002e658ef80524180720e3fe73572d76390fe4baff708d5c7546f972aff825e5377a536804488dc933720800.

Reply:
Well.. now I think about it, if it passes CI then it should be good!